### ISSUE TITLE
DNSSEC false positives

### ISSUE THREAD

**Issue by japaric** [hickory-dns/hickory-dns#2389]
**Describe the bug**
These are scenarios where hickory-dns sets the AD flag in the response when it ought not

**To Reproduce**
Send queries of the form `A $SUB.extended-dns-errors.com.`  to hickory where `$SUB` is one of these

- bad-ksk
- bad-rrsig-dnskey
- bad-rrsig-ksk
- ds-bad-key-algo
- ds-bad-tag
- ds-bogus-digest-value
- ds-reserved-key-algo
- ds-unassigned-key-algo
- no-dnskey-257
- no-ds
- no-ksk
- no-rrsig-dnskey
- no-rrsig-ksk

**Expected behavior**

In the case of these subdomains `ds-reserved-key-algo`, `ds-unassigned-key-algo`, `no-ds`, unbound responds with NOERROR and the AD flag unset. In the other cases, unbound responds with SERVFAIL.

hickory should behave similarly and definitively should not set the AD flag in the response.

**System:**
 - OS: Ubuntu
 - Architecture: x86_64
 - Version 24.04
 - rustc version: 1.80.1

**Version:**
Crate: `hickory-dns`
Version: 77bd54e2ca98973b40210f20772d5f7f2de18143

**Additional context**
Results produced using the test suite that's being added in #2385

**Comment by djc** [comment]
Still two checkboxes left, should this be kept open for now?

**Comment by japaric** [comment]
no, those last two were fixed in #2399 : https://github.com/hickory-dns/hickory-dns/pull/2399/files#diff-59514c0e8f8e69369efb4c2ce32d3f0b9a9a5ba2db81351215b8c8971644ce56
